### PR TITLE
[FIX] account: quick edit analytic distribution posted line

### PR DESCRIPTION
**Pull request by guva-odoo**
With this commit we allow to change/add analytic distribution
on a posted line in quick edit mode.

opw-3561831

### CODE REVIEW

**Review by gawa-odoo — APPROVED** [review]
LGTM

**Review by Levizar — APPROVED** [review]
@robodoo r+

File: addons/account/tests/test_account_analytic.py [hunk 1]
```diff
@@ -210,3 +210,55 @@ def test_mandatory_plan_validation(self):
         invoice.invoice_line_ids.analytic_distribution = {self.analytic_account_b.id: 0.9}
         invoice.action_post()
         self.assertEqual(invoice.state, 'posted')
+
+    def test_set_anaylytic_distribution_posted_line(self):
```
Comment (Levizar) [inline addons/account/tests/test_account_analytic.py:214]: `anaylytic` I just see it now :rofl:

**Comment by ivantodorovich** [comment]
Hello @guva-odoo 

The PR description suggests the user would be allowed to add _or change_ analytic distribution.
However, in our case, we see users are only allowed to add, but not change.

e.g.: this adapted unit test fails:

```diff
     def test_set_analytic_distribution_posted_line(self):
         """
         Test that we can set the analytic distribution on the product line of a move, when the line has tax with
         repartition lines used in tax closing. Although the change can not be applied on the tax line, we should
         not raise any error.
         """
         tax = self.tax_purchase_a.copy({
             'name': 'taXXX',
             'invoice_repartition_line_ids': [
                 Command.create({
                     'repartition_type': 'base',
                     'use_in_tax_closing': False,
                 }),
                 Command.create({
                     'repartition_type': 'tax',
                     'factor_percent': 50,
                     'use_in_tax_closing': False,
                 }),
                 Command.create({
                     'repartition_type': 'tax',
                     'factor_percent': 50,
                     'account_id': self.company_data['default_account_tax_purchase'].id,
                     'use_in_tax_closing': True,
                 }),
             ],
             'refund_repartition_line_ids': [
                 Command.create({
                     'repartition_type': 'base',
                     'use_in_tax_closing': False,
                 }),
                 Command.create({
                     'repartition_type': 'tax',
                     'factor_percent': 50,
                     'use_in_tax_closing': False,
                 }),
                 Command.create({
                     'repartition_type': 'tax',
                     'factor_percent': 50,
                     'account_id': self.company_data['default_account_tax_purchase'].id,
                     'use_in_tax_closing': True,
                 }),
             ],
         })

         bill = self.create_invoice(self.partner_a, self.product_a, move_type='in_invoice')
         bill.invoice_line_ids.tax_ids = [Command.set(tax.ids)]
+        line = bill.line_ids.filtered(lambda l: l.display_type == 'product')
+
+        # set distribution before posting
+        line.write({'analytic_distribution': {self.analytic_account_a.id: 50}})
+
+        # post the bill
         bill.action_post()

+        # modify distribution
-        line = bill.line_ids.filtered(lambda l: l.display_type == 'product')
         line.write({'analytic_distribution': {self.analytic_account_a.id: 100}})
         self.assertEqual(line.analytic_distribution, {str(self.analytic_account_a.id): 100})
```

```
Traceback (most recent call last):
  File "/odoo/src/addons/account/tests/test_account_analytic.py", line 269, in test_set_analytic_distribution_posted_line
    line.write({'analytic_distribution': {self.analytic_account_a.id: 100}})
  File "/odoo/src/addons/account/models/account_move_line.py", line 1530, in write
    line.move_id._message_log(
  File "/usr/lib/python3.9/contextlib.py", line 124, in __exit__
    next(self.gen)
  File "/odoo/src/addons/account/models/account_move.py", line 2211, in _sync_dynamic_lines
    update_containers()
  File "/usr/lib/python3.9/contextlib.py", line 513, in __exit__
    raise exc_details[1]
  File "/usr/lib/python3.9/contextlib.py", line 135, in __exit__
    self.gen.throw(type, value, traceback)
  File "/odoo/src/addons/account/models/account_move.py", line 2079, in _sync_dynamic_line
    yield
  File "/usr/lib/python3.9/contextlib.py", line 498, in __exit__
    if cb(*exc_details):
  File "/usr/lib/python3.9/contextlib.py", line 124, in __exit__
    next(self.gen)
  File "/odoo/src/addons/account/models/account_move.py", line 2133, in _sync_dynamic_line
    self.env['account.move.line'].browse(line_id).write(
  File "/odoo/src/addons/account/models/account_move_line.py", line 1473, in write
    raise UserError(_('You cannot modify the taxes related to a posted journal item, you should reset the journal entry to draft to do so.'))
odoo.exceptions.UserError: You cannot modify the taxes related to a posted journal item, you should reset the journal entry to draft to do so.
```

I would love to have more context on that opw, in order to know if this is a design decision or a bug 😓
Is it possible for external contributors like me to peek inside opw's?

**Comment by gawa-odoo** [comment]
@ivantodorovich Hello!
Indeed, the fix is not working if you already have a distribution. The ticket was about a simple case where they wanted to add a distribution, so it wasn't noticed.
It's definitely a bug. We noticed recently that we have other issues with it (you can't change the account_d, or the vehicle_id with this kind of tax).
We've created a task to tackle it

**Comment by ivantodorovich** [comment]
Thanks for the reply, @gawa-odoo ❤️

**Comment by Levizar** [comment]
@guva-odoo what do ze do about those fw port?

**Pull request closed without merging**